pkg/controller/deployment/deployment_controller.go

```diff
@@ -482,7 +482,7 @@ func (dc *DeploymentController) handleErr(err error, key interface{}) {
 	}
 
 	if dc.queue.NumRequeues(key) < maxRetries {
-		klog.V(2).Infof("Error syncing deployment %v: %v", key, err)
+		klog.ErrorS(err, "Error syncing deployment", "deployment", key)
```

Comment:
1. Let's use `klog.V(2).InfoS` and add `err` as key.
    I should make it more clearer, but we should not change `Info` to `Error` as this changes verbosity 
2. Can you trace in code what is expected value of `key`? We should avoid writing `interface{}` type under `deployment` key.

Reply:
key is `name/ns`